packages/fluentui/react-northstar/src/components/Header/Header.tsx

```diff
@@ -38,56 +44,86 @@ export interface HeaderProps
 }
 
 export const headerClassName = 'ui-header';
+export type HeaderStylesProps = Required<Pick<HeaderProps, 'align' | 'color'>> & { hasDescription: boolean };
+
+const Header: React.FC<WithAsProp<HeaderProps>> &
+  FluentComponentStaticProps<HeaderProps> & {
+    Description: typeof HeaderDescription;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Header.displayName, context.telemetry);
+  setStart();
+  const { children, content, variables, align, className, design, styles, description, color } = props;
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Header.handledProps, props);
+  const hasChildren = childrenExist(children);
+  const contentElement = childrenExist(children) ? children : content;
+
+  const getA11yProps = useAccessibility<never>(props.accessibility, {
+    debugName: Header.displayName,
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<HeaderStylesProps>(Header.displayName, {
+    className: headerClassName,
+    mapPropsToStyles: () => ({
+      align,
+      hasDescription: !!description,
+      color,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  const element = (
+    <ElementType
+      {...getA11yProps('root', {
+        className: classes.root,
+        ...unhandledProps,
+        ...rtlTextContainer.getAttributes({
+          forElements: [children, content],
+          condition: !description,
+        }),
+      })}
+    >
+      {rtlTextContainer.createFor({ element: contentElement, condition: !!description })}
+      {!hasChildren &&
+        HeaderDescription.create(description, {
+          defaultProps: () =>
+            getA11yProps('description', {
+              variables: {
+                ...(variables?.descriptionColor && { color: variables.descriptionColor }),
```

Comment:
Oops, previously with has `resolvedVariables` so these things are not equal. I suggest to drop this statement and it as a breaking change.

Reply:
Dunno what you mean, if we removed this statement our own example setting the `descriptionColor` wouldn't work. Do you mean to us to se a new prop `DescriptionColor?: string` to `Header`?

Reply:
https://github.com/microsoft/fluentui/blob/c3ed69100489a037d5e8def9296fe1055ea236c6/packages/fluentui/react-northstar/src/components/Header/Header.tsx#L62

1. In class component `variables` are coming from `renderComponent()`
2. `variables` from `renderComponent()` are `resolvedVariables`
  - are always `Object`
  - will include all `headerVariables`: https://github.com/microsoft/fluentui/blob/master/packages/fluentui/react-northstar/src/themes/teams/components/Header/headerVariables.ts
3. It means that `variables` from `renderComponent()` are not equal to `props.variables`

***

My suggestion is to remove this statement at all:

```suggestion
```

Reply:
And please add this as a breaking change